### DISCUSSION TITLE
Create @scope Blade directives which creates a variable scope like @include but does not require a file.

### DISCUSSION THREAD

**Discussion by christianmagill** [laravel/framework#38844] [Ideas]
Often I find myself wanting to `extract()` the contents of an array for ease of coding, but I don't want the data polluting the global space.

I'd love to see `@scope` and `@endscope` directives added for times when scoping would be useful, but an `@include` wouldn't make sense.

Perhaps the first parameter takes an array of data to be extracted, and a second optional parameter would determine if the view's scope should be inherited or not.

So a simple example might look like

```
@foreach($links as $link)
   @scope($link, false)
      {{-- 
         Because inheriting scope of the view is set to false 
         there are no worries about variables existing from previous code.
         Likewise, we don't have to worry about these extracted values
         polluting the code to be executed down the line.
      --}}
      <a href="{{ $url ?? '' }}" target="{{ $taret ?? '' }}">{{ $title ?? '' }}<a>
   @endscope
@endforeach

```

**Comment by shaedrich** [comment]
For reference: Other templating engines implemented it like this:
* [Handlebars](https://handlebarsjs.com/guide/#evaluation-context)
* [Twig](https://twig.symfony.com/doc/3.x/tags/with.html)
